tests/providers/amazon/aws/executors/ecs/test_ecs_executor.py

```diff
@@ -1086,3 +1089,31 @@ def test_start_health_check_config(self, set_env_vars):
         executor.start()
 
         ecs_mock.stop_task.assert_not_called()
+
+    def test_providing_both_capacity_provider_and_launch_type_fails(self, set_env_vars):
+        os.environ[
+            f"AIRFLOW__{CONFIG_GROUP_NAME}__{AllEcsConfigKeys.CAPACITY_PROVIDER_STRATEGY}".upper()
+        ] = "[{'capacityProvider': 'cp1', 'weight': 5}, {'capacityProvider': 'cp2', 'weight': 1}]"
+        expected_error = (
+            "capacity_provider_strategy and launch_type are mutually exclusive, you can not provide both."
+        )
+
+        with pytest.raises(ValueError, match=expected_error):
+            AwsEcsExecutor()
+
+    def test_providing_capacity_provider(self, set_env_vars):
+        valid_capacity_provider = (
+            "[{'capacityProvider': 'cp1', 'weight': 5}, {'capacityProvider': 'cp2', 'weight': 1}]"
+        )
+
+        os.environ[
+            f"AIRFLOW__{CONFIG_GROUP_NAME}__{AllEcsConfigKeys.CAPACITY_PROVIDER_STRATEGY}".upper()
+        ] = valid_capacity_provider
+        os.environ.pop(f"AIRFLOW__{CONFIG_GROUP_NAME}__{AllEcsConfigKeys.LAUNCH_TYPE}".upper())
```

Comment:
We should add two more test cases where you don't provide a capacity provider and also remove this launch type config. And ensure that the two layer defaulting works (one test where the cluster default is used and another where the ultimate fallback of fargate launch type is used).

Another option is to no longer set the launch type in `set_env_vars` and let the defaulting put it in, then you wouldn't need to test that specific case, since it'd be tested everywhere (but you still need one for the cluster default in that case).

Reply:
Tests added (pending static checks and CI passing locally) thanks for the suggestion.

Reply:
Added the missing tests and a one-line comment to each to help make it more obvious what is being tested, the test names were getting pretty convoluted.